### PR TITLE
[CD] enable s3 publish for nightly builds in cd

### PR DESCRIPTION
**Pull request by szha**
## Description ##
enable s3 publish for nightly builds in cd

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] enable publish to s3 from cd pipeline

### CODE REVIEW

**Comment by szha** [comment]
@perdasilva @samskalicky @leezu

File: cd/python/pypi/pypi_publish.py [hunk 1]
```diff
@@ -47,14 +48,14 @@ def post_wheel(path):
         print('In DEV account, skipping publish')
         print('Would have run: {}'.format(cmd))
         return 0
-    else:
+    elif any(test_version_mark in version for test_version_mark in ['a', 'b', 'dev']):
```
Comment (leezu) [inline cd/python/pypi/pypi_publish.py]: How is this condition related to `See https://github.com/pypa/pypi-support/issues/50 for details`?

Comment (szha) [reply cd/python/pypi/pypi_publish.py]: only rc is allowed for pre-releases.

Comment (leezu) [reply cd/python/pypi/pypi_publish.py]: You can consider adding a comment to clarify.

File: cd/python/pypi/pypi_publish.py [hunk 2]
```diff
@@ -47,14 +48,14 @@ def post_wheel(path):
...
         print('Skipping publishing nightly builds to Pypi.')
         print('See https://github.com/pypa/pypi-support/issues/50 for details')
         return 0
 
+    else:
         # DO NOT PRINT CMD IN THIS BLOCK, includes password
-        p = subprocess.run(cmd.split(' '),
-                        stdout=subprocess.PIPE)
+        p = subprocess.run(cmd.split(' '), stdout=subprocess.PIPE)
```
Comment (marcoabreu) [inline cd/python/pypi/pypi_publish.py]: This is insecure because this will print the password in case of an exception. Passwords should always be passed using stdin

Comment (szha) [reply cd/python/pypi/pypi_publish.py]: twine doesn't have the option to pass credentials through stdin. I updated this part to pass credentials through environment variable in this process instead.

File: cd/python/pypi/pypi_publish.py [hunk 3]
```diff
@@ -34,11 +34,11 @@ def post_wheel(path):
     """
     logging.info('Posting {} to PyPI'.format(path))
     pypi_credentials = get_secret()
+    os.environ['TWINE_USERNAME'] = pypi_credentials['username']
+    os.environ['TWINE_PASSWORD'] = pypi_credentials['password']
```
Comment (marcoabreu) [inline cd/python/pypi/pypi_publish.py]: Can you unset them in a final block? Sometimes we print env vars for debugging purposes in case of errors. Don't want them to be leaked

Comment (szha) [reply cd/python/pypi/pypi_publish.py]: Instead of writing to the environment variable of current process, I moved the credentials to the subprocess environment variable.

Comment (marcoabreu) [reply cd/python/pypi/pypi_publish.py]: Even better, great idea!
